Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -3427,11 +3417,10 @@ public List<Permanent> getAvailableManaProducersWithCost(Game game) {
      * @param ability
      * @param availableMana if null, it won't be checked if enough mana is
      *                      available
-     * @param sourceObject
      * @param game
      * @return
      */
-    protected boolean canPlay(ActivatedAbility ability, ManaOptions availableMana, MageObject sourceObject, Game game) {
+    protected boolean canPlay(ActivatedAbility ability, ManaOptions availableMana, Game game) {
```

Comment:
See above about source and game params removes. `sourceObject` is not used here after alternative cost fixes in #9049, but I don't like the idea to remove object's info. It's can be useful for future improvements or fixes (`ability->sourceId` and `sourceObject` are different for mdf/dfc/etc cards with multi faces, e.g. shared abilities).

![shot_230715_104833](https://github.com/magefree/mage/assets/8344157/9557fb9b-293e-4469-b4d8-70270136673a)